src/mixins/object_geometry.mixin.js

```diff
@@ -187,6 +187,35 @@
       return false;
     },
 
+    /**
+     * Checks if object is partially contained within the canvas with current viewportTransform
+     * the check is done stopping at first point that appears off screen
+     * @param {Boolean} [calculate] use coordinates of current position instead of .oCoords
```

Comment:
use coordinates of current position, calculated just in time, instead of precalculated .aCoords

Reply:
apply to the other function too.